platform/mbed_wait_api_rtos.cpp

```diff
@@ -23,26 +23,46 @@
 #include "rtos/rtos.h"
 #include "platform/mbed_critical.h"
 #include "platform/mbed_power_mgmt.h"
+#include "platform/mbed_error.h"
 
 void wait(float s)
 {
-    wait_us(s * 1000000.0f);
+    if (s >= 0.01f) {
+        wait_ms(s * 1000.0f);
```

Comment:
If wait is used from an interrupt handler it will only trap if the delay is `>= 10ms`. Is this acceptable?

Reply:
Fixed.. wait will not trap